File: frontend/src/app/models/communityProfile/CommunityProfileDTO.ts
import { RoleDTO } from '../role/RoleDTO';
import { TitleDTO } from '../title/TitleDTO';

export interface CommunityProfileDTO {
  id: number;
  username: string;
  description: string | null;
  profileImage: string | null;
  featuredTitle: TitleDTO | null;
  userId: number;
  communityId: number;
  roles: RoleDTO[];
  titles: TitleDTO[];
}

File: backend/src/main/java/com/guildnet/backend/features/communityProfile/dto/CommunityProfileDTO.java
package com.guildnet.backend.features.communityProfile.dto;

import com.guildnet.backend.features.role.dto.RoleDTO;
import com.guildnet.backend.features.title.Title;
import com.guildnet.backend.features.title.dto.TitleDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CommunityProfileDTO {
    private Long id;
    private String username;
    private String description;
    private String profileImage;
    private TitleDTO featuredTitle;
    private Long userId;
    private Long communityId;
    private List<RoleDTO> roles;
    private List<TitleDTO> titles;
}
